lua/gitlab/git.lua

```diff
@@ -43,6 +43,60 @@ M.switch_branch = function(branch)
   return run_system({ "git", "checkout", "-q", branch })
 end
 
+---Fetches the name of the remote tracking branch for the current branch
+---@return string|nil, string|nil
+M.get_remote_branch = function()
+  return run_system({ "git", "rev-parse", "--abbrev-ref", "--symbolic-full-name", "@{u}" })
+end
+
+---Determines whether the tracking branch is ahead of or behind the current branch, and warns the user if so
+---@param current_branch string
+---@param remote_branch string
+---@return boolean
+M.get_ahead_behind = function(current_branch, remote_branch, log_level)
+  local u = require("gitlab.utils")
+  local result, err =
+    run_system({ "git", "rev-list", "--left-right", "--count", current_branch .. "..." .. remote_branch })
+  if err ~= nil or result == nil then
+    u.notify("Could not determine if branch is up-to-date: " .. err, vim.log.levels.ERROR)
+    return false
+  end
+
+  local ahead, behind = result:match("(%d+)%s+(%d+)")
+  if ahead == nil or behind == nil then
+    u.notify("Error parsing ahead/behind information.", vim.log.levels.ERROR)
+    return false
+  end
+
+  ahead = tonumber(ahead)
+  behind = tonumber(behind)
+
+  if ahead > 0 and behind == 0 then
+    u.notify(
+      string.format("You have local commits that are not on %s. Have you forgotten to push?", remote_branch),
```

Comment:
Maybe you could make this consistent with the other warning:
```suggestion
      string.format("There are local changes that haven't been pushed to %s yet", remote_branch),
```